src/main/java/org/tron/common/runtime/vm/program/Program.java

```diff
@@ -88,6 +88,25 @@
   //Max size for stack checks
   private static final int MAX_STACKSIZE = 1024;
 
+  public static byte[] getRootTransactionId() {
+    return rootTransactionId;
+  }
+
+  public static void setRootTransactionId(byte[] rootTransactionId) {
+    Program.rootTransactionId = rootTransactionId;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "rootTransactionId". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1341&issues=AWVhvRIAqLxyBbb_7qtG&open=AWVhvRIAqLxyBbb_7qtG)

[ISSUE_KEY]: AWVhvRIAqLxyBbb_7qtG